netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -323,10 +323,13 @@ void returnProcessedBytes(Http2Stream stream, int bytes) {
   }
 
   private void onHeadersRead(int streamId, Http2Headers headers, boolean endStream) {
-    NettyClientStream.TransportState stream = clientStream(requireHttp2Stream(streamId));
-    stream.transportHeadersReceived(headers, endStream);
-    if (keepAliveManager != null) {
-      keepAliveManager.onDataReceived();
+    // Stream 1 is reserved for the Upgrade response, so we should ignore its headers here:
+    if (streamId != 1) {
```

Comment:
It seems like this should check the negotiation type before ignoring the headers.

Reply:
@carl-mastrangelo, how do you suggest checking the negotiation type? At present, Netty reserves streamId 1 only for upgrades.

Reply:
The comment made it seem like it was more general than Netty.  Maybe the comment should be updated?